src/activities/ascending_order/ascending_order.js

```diff
@@ -0,0 +1,287 @@
+/* GCompris - ascending_order.js
+ *
+ * Copyright (C) 2016 Rudra Nil Basu <rudra.nil.basu.1996@gmail.com>
+ *
+ * Authors:
+ *   Rudra Nil Basu <rudra.nil.basu.1996@gmail.com> (Qt Quick port)
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+.pragma library
+.import QtQuick 2.0 as Quick
+
+var currentLevel = 0
+var numberOfLevel = 4
+
+var items
+var mode
+
+// num[] will contain the random numbers
+var num = []
+var originalArrangement = []
+var alphabets = []
+
+var ascendingOrder
+var thresholdDistance
+var noOfTilesInPreviousLevel
+
+function start(items_, mode_) {
+    items = items_
+    mode = mode_
+    currentLevel = 0
+    items.flow.validMousePress = false
+    thresholdDistance = 4000 * items.ratio
+    items.score.currentSubLevel = 0
+    items.score.numberOfSubLevels = 4
+
+    if (mode == "alphabets") {
+        //: list containing all the characters separated by a "/"
+        var letters = qsTr("a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/q/r/s/t/u/v/w/x/y/z")
+        alphabets = letters.split("/")
+    }
+
+    noOfTilesInPreviousLevel = -1
+    initLevel()
+}
+
+function stop() {
+}
+
+function initLevel() {
+    ascendingOrder = items.score.currentSubLevel % 2 == 0 ? true : false
```

Comment:
to be checked on the mailing list (or at least with Emmanuel) but I wouldn't change every sublevel. It would not be efficient to understand the meaning behind ascending/descending order